scrubber/core.py

```diff
@@ -565,78 +565,90 @@ def __call__(self, input_mol):
 
         return output_mol_list
     
-def ConstrainedEmbeding(query_mol, core_mol, confId=-1, randomseed=2342, template_smarts=None,
-                     ff='uff'):
-  """ Generate an embedding of a query molecule where part of the molecule
+def constrained_embeding(
+    query_mol,
+    core_mol,
+    template_smarts: str = None,
+    numconfs: int = 1,
+    ff: str = "mmff94s",
+    ps=None,
+):
+    """Generate an embedding of a query molecule where part of the molecule
     is constrained to have particular coordinates derived from a core.
-    Alternatively, a SMARTs pattern can be provided to match specific atoms from both molecules 
-  """
-  force_constant = 1000
-
-  if ff == 'uff' or ff == 'espaloma':
-    getForceField=AllChem.UFFGetMoleculeForceField
-  elif ff == 'mmff94':
-    getForceField=lambda x:AllChem.MMFFGetMoleculeForceField(x,AllChem.MMFFGetMoleculeProperties(x),confId=confId)
-  elif ff == 'mmff94s':
-    getForceField=lambda x:AllChem.MMFFGetMoleculeForceField(x,AllChem.MMFFGetMoleculeProperties(x, mmffVariant='MMFF94s'),confId=confId)
-
-  if template_smarts == None:
-    query_match = query_mol.GetSubstructMatches(core_mol)
-    if not query_match:
-      raise ValueError("molecule doesn't match the core")
-    elif len(query_match) > 1:
-      raise RuntimeError('Expected one match but multiple matches were found.')
-    else:
-      query_match=query_match[0]
+    Alternatively, a SMARTs pattern can be provided to match specific atoms from both molecules
+    """
 
-    algMap = [(j, i) for i, j in enumerate(query_match)]
-    
-  else:
-    query_match = query_mol.GetSubstructMatches(template_smarts)
-    
-    if not query_match:
-      raise ValueError("SMARTs doesn't match the molecule")
-    elif len(query_match) > 1:
-      raise RuntimeError('Expected one match but multiple matches were found.')
-    else:
-      query_match = query_match[0]
+    force_constant = 1000
+    confId = -1
+
+    if ff == "uff":
+        getForceField = AllChem.UFFGetMoleculeForceField
+    elif ff == "mmff94" :
+        getForceField = lambda x: AllChem.MMFFGetMoleculeForceField(
+            x, AllChem.MMFFGetMoleculeProperties(x), confId=confId
+        )
+    elif ff == "mmff94s" or ff == "espaloma":
```

Comment:
yes I know, but this is a minimization with restraints to force the querry mol to match the template. If you chose espaloma as ff it will still minimize it at the end after this stage